inspirehep/modules/hal/tei.py

```diff
@@ -0,0 +1,177 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+from elasticsearch import RequestError
+from nameparser import HumanName
+
+from inspirehep.dojson.utils import get_recid_from_ref
+from inspirehep.modules.records.json_ref_loader import replace_refs
+from inspirehep.utils.dedupers import dedupe_list
+from inspirehep.utils.record import get_value
+from inspirehep.utils.record_getter import get_es_records
+
+from .templates import template
+
+
+def tei_response(record):
+    """Returns the record formatted in XML+TEI per HAL's specification.
+
+    Keyword arguments:
+    record -- the record to convert
```

Comment:
I don't remember if here we agreed to keep it like this or to try using Sphinx's style.